datastore/google/cloud/datastore/entity.py

```diff
@@ -181,6 +181,19 @@ def kind(self):
         if self.key:
             return self.key.kind
 
+    @property
+    def id(self):
+        """Get the id of the current entity.
+
+        .. note::
+          This relies entirely on the :class:`gcloud.datastore.key.Key`
+          set on the entity.  That means that we're not storing the id
+          of the entity at all, just the properties and a pointer to a
+          Key which knows its Id.
+        """
+        if self.key:
```

Comment:
Please use an explicit `None`-check: `if self.key is not None`

![nunchucks-nuns-chuck-norris-nun-12593488599](https://user-images.githubusercontent.com/520669/29418477-41fdd4e0-8321-11e7-86c9-f0f44f0225d6.jpg)